<!doctype html>
<html lang="en">

<head>
    <title>Reports - Weekly update 23/06/07</title>
  <meta charset="utf-8" />
  <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">



    <meta name="tags" content="peptide_pooling" />
    <meta name="tags" content="public_tcrs" />
    <meta name="tags" content="hgt" />

  <link rel="stylesheet" href="./theme/css/bootstrap.min.css" />
  <link rel="stylesheet" href="./theme/css/friendly.css" /> <!-- Adapted from https://richleland.github.io/pygments-css/ -->
  <link rel="stylesheet" href="./theme/css/custom.css" />
  <style media="screen">@font-face {font-family: ubuntumono;src: url("./theme/css/UbuntuMono-Regular.ttf");}</style>
</head>

<body id="index" class="home">

  <nav class="navbar navbar-expand-lg navbar-dark bg-dark">
    <a class="navbar-brand" href="https://meyerdiaries.cshl.edu/">The Meyer Diaries Homepage</a>

    <button class="navbar-toggler" type="button" data-toggle="collapse" data-target="#navbarSupportedContent">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div class="navbar collapse navbar-collapse" id="navbarSupportedContent">
      <ul class="navbar-nav">
        <li class="nav-item">
            <a href="." class="nav-link">Reports</a>
        </li>
      </ul>
      <ul class="navbar-nav">


        <li class="nav-item" >
          <a href="./category/plans.html" class="nav-link">plans</a>
        </li>
        <li class="nav-item active" >
          <a href="./category/updates.html" class="nav-link">updates</a>
        </li>
      </ul>
      <ul class="navbar-nav">
        <li class="nav-item">
            <a href="./everything.html" class="nav-link">By date</a>
        </li>
      </ul>
    </div>
  </nav><!-- /#menu -->

  <div class="container main-content" style="margin-bottom: 200px">
<section id="content" class="body">
  <header>
    <h2 class="entry-title">
      <a href="./weekly-update-230607.html" rel="bookmark" title="Permalink to Weekly update 23/06/07">Weekly update 23/06/07</a>
    </h2>
  </header>
  <footer class="post-info">
    <p>
      Created: <time class="published" datetime="2023-06-07T10:00:00+02:00">Wed 07 June 2023</time>
    </p>
    Category: <a href="./category/updates.html">updates</a>
    <div class="tags">
        Tags:
            <a href="./tag/peptide_pooling.html">peptide_pooling</a>
            <a href="./tag/public_tcrs.html">public_tcrs</a>
            <a href="./tag/hgt.html">hgt</a>
    </div>
  </footer><!-- /.post-info -->
  <div class="entry-content">
    <hr>
    <h1>Peptide pooling</h1>
<h3>Outline</h3>
<p>Algorithmic tasks:</p>
<ul>
<li>
<p>drop-outs (especially useful for the PBMCs sequencing part)</p>
</li>
<li>
<p>improve address selection using Hannahâ€™s idea â€” <strong><a href="https://en.wikipedia.org/wiki/Set_cover_problem">Set coverage problem</a></strong></p>
</li>
<li>
<p>balance between number of pools, number of peptides in one pool, using userâ€™s threshold</p>
</li>
<li>
<p>add to the code option to choose the number of max undistinguishable peptides (for more overlapping peptides)</p>
</li>
<li>
<p>add to the code check for how strongly peptides overlap</p>
</li>
</ul>
<h3>Drop-outs</h3>
<h4>Task description</h4>
<ul>
<li>Possible mistakes:<ul>
<li>drop-outs, causing:<ul>
<li>impossibility of narrowing down peptide options to 2 peptides</li>
<li>absence of right peptide in possible options</li>
</ul>
</li>
<li>additions, causing:<ul>
<li>impossibility of narrowing down peptide options to 2 peptides</li>
</ul>
</li>
<li>drop-outs AND additions<ul>
<li>impossibility of narrowing down peptide options to 2 peptides</li>
<li>absence of right peptide in possible options</li>
</ul>
</li>
</ul>
</li>
<li>Impossibility of narrowing down peptide options to 2 peptides can be solved (or at least minimized):<ul>
<li>by redundancy of addresses</li>
<li>by some general rule of address generation (i.e. sum of all addresses is even or something like that)</li>
</ul>
</li>
<li>Absence of right peptide in possible options can be solved:<ul>
<li>by determination when and how many pools were dropped, for example by creating a general rule for the number of activated pools</li>
</ul>
</li>
</ul>
<aside>
ðŸ’¡ Conclusion. We need:
1) rule for the determination of when and how many pools were lost (i.e. # of activated pools should always be 5, or 
2) rule for the pools in the address to be able to decrease number of possible options (by the way, it will lead to redundancy as well â€” because not all addresses will be available)

</aside>

<h4>Tasks</h4>
<ul>
<li>[x]  find a metric for algorithm accuract in presence of dropouts (# of peptides M / # of dropouts looks nice, but maybe needs a little more rethinking); # of event / # of peptides M plot needs rethinking too</li>
<li>[x]  I tried a simple algorithm (# of pools â‰¤ ITERS +1 AND sum(pools)%2 == 0), but there is not enough addresses for this solution â†’ I need to look at the peptides and calculate on paper how these two ideas work together</li>
<li>[ ]  Think about different version of address assignment: I need to ensure that epitopes lying on the intersection of two peptides have unique address â€”&gt; I do not need to think about addresses for peptides but rather addresses (i.e. activated pools) for epitopes</li>
</ul>
<h4>Results</h4>
<h5>How drop-outs are handled now without any changes to the algorithm</h5>
<p><img alt="Metric 1: All possible dropouts starting from two activated pools are shown. One event = activation if pools with one epitope with 0 to ITERS-2 drop-outs" src="./images/current_algorithm_m1.png">
All possible dropouts starting from two activated pools are shown. One event = activation if pools with one epitope with 0 to ITERS-2 drop-outs</p>
<p><img alt="Metric 2" src="./images/current_algorithm_m2.png">
How number of drop-outs impacts number of possible peptides</p>
<h5>Altered algorithm</h5>
<ul>
<li>'# of activated pools == iters +1</li>
<li>sum(address) is even</li>
</ul>
<p>To satisfy the first requirement, rearrangement of addresses was needed.
Like this:</p>
<div class="highlight"><pre><span></span><code>(0, 1, 2, 3)
(0, 1, 2, 4)
(0, 1, 2, 5)
(0, 1, 2, 6)
(0, 1, 2, 7)
(0, 1, 2, 8)
(0, 1, 2, 9)
(0, 1, 2, 10)
(0, 1, 2, 11)
(0, 1, 3, 4)
(0, 1, 3, 5)
(0, 1, 3, 6)
(0, 1, 3, 7)
(0, 1, 3, 8)
(0, 1, 3, 9)
(0, 1, 3, 10)
(0, 1, 3, 11)
(0, 1, 4, 5)
(0, 1, 4, 6)
(0, 1, 4, 7)
(0, 1, 4, 8)
...
</code></pre></div>

<div class="highlight"><pre><span></span><code>(0, 1, 2, 3)
(0, 1, 2, 4)
(0, 1, 2, 5)
(0, 1, 2, 6)
(0, 1, 2, 7)
(0, 1, 2, 8)
(0, 1, 2, 9)
(0, 1, 2, 10)
(0, 1, 2, 11)
(0, 1, 3, 11)
(0, 1, 3, 10)
(0, 1, 3, 9)
(0, 1, 3, 8)
(0, 1, 3, 7)
(0, 1, 3, 6)
(0, 1, 3, 5)
(0, 1, 3, 4)
(0, 1, 4, 5)
(0, 1, 4, 6)
(0, 1, 4, 7)
(0, 1, 4, 8)
...
</code></pre></div>

<p>And then addresses with odd sum were filtered out. Unfortunately, it should be an iterative procedure (+ getting rid of the bad addresses), and because of that a lot of addresses were lost.</p>
<p>This scheme turned out to be more susceptible to collisions, because now similar addresses have similar neighbours:</p>
<ul>
<li>(0, 1, 2, 10) â€” <code>VYYPDKVFRSSVLHSTQ</code></li>
<li>(0, 1, 2, 11) â€” <code>KVFRSSVLHSTQDLFLP</code></li>
<li>(0, 1, 10, 11) â€” <code>SALEPLVDLPIGINITR</code></li>
<li>(0, 2, 10, 11) â€” <code>VYYPDKVFRSSVLHSTQ</code></li>
</ul>
<p>In previous algorithms such cases were rare, because similar addresses (0, 1, 2, 10) and (0, 1, 10, 11) have very different neighbours (0, 1, 2, 11) and (0, 2, 3, 4). As a result, a lot of possible addresses are filtered out.</p>
<table>
<thead>
<tr>
<th>N_POOLS</th>
<th>ITERS</th>
<th>Initial number of addresses</th>
<th>Organized addreses with even sum without collisions</th>
</tr>
</thead>
<tbody>
<tr>
<td>12</td>
<td>3</td>
<td>220</td>
<td>80</td>
</tr>
<tr>
<td>12</td>
<td>4</td>
<td>495</td>
<td>42</td>
</tr>
<tr>
<td>13</td>
<td>3</td>
<td>286</td>
<td>88</td>
</tr>
<tr>
<td>13</td>
<td>4</td>
<td>715</td>
<td>77</td>
</tr>
<tr>
<td>14</td>
<td>3</td>
<td>364</td>
<td>122</td>
</tr>
<tr>
<td>18</td>
<td>3</td>
<td>816</td>
<td>309</td>
</tr>
<tr>
<td>18</td>
<td>4</td>
<td>3060</td>
<td>214</td>
</tr>
<tr>
<td>18</td>
<td>5</td>
<td>8568</td>
<td>184</td>
</tr>
</tbody>
</table>
<p>But this scheme solved the proble of the absence of the TRUE peptide in possible peptides.</p>
<p><img alt="Metric 1" src="./images/alt1_algorithm_m1.png"></p>
<p><img alt="Metric 2" src="./images/alt1_algorithm_m2.png"></p>
<p>However, number of possible peptides skyrocketed. That is because number of pools was very high (18).</p>
<p>Other problems:</p>
<ul>
<li>clever mechanism of pools balance will be needed to not disturb special address rearrangement when all two neighbour addresses has sum of set == ITERS +1</li>
<li>too many pools, and because of that algorithm is clearly non-scalable</li>
<li>still determination of TRUE peptide is problematic again because of too many pools</li>
</ul>
<h5>Only # of activated pools were restricted</h5>
<p>Then I tried using only number of activated pools restriction, without any particular rule for the address selection (even sum), because even sum ruled out a lot of possible addresses (near a half).</p>
<p><img alt="Metric 1" src="./images/alt1.2_algorithm_m1.png"></p>
<p><img alt="Metric 2" src="./images/alt1.2_algorithm_m2.png"></p>
<h1>Public TCRs</h1>
<h3>Outline</h3>
<ul>
<li>I need to read a paper about TCRdb â€” why do they have beta chains there only?</li>
<li>I need to download information about alpha-chains from projects myself (maybe I can use TCRdb workflow of database curation?)</li>
<li>Pathological conditions need to be manually curated</li>
<li>paper on TCR fingerprinting clustering</li>
<li>paper about CDR3 length â€” <a href="https://www.sciencedirect.com/science/article/pii/S2589004222012408">Two types of human TCR differentially regulate reactivity to self and non-self antigens</a></li>
<li>Why are TRA CDR3s in McPAS shorter than others?</li>
<li>Clustering tool â€” <a href="https://academic.oup.com/bioinformatics/article/37/24/4865/6300511">ClusTCR</a> (<a href="https://svalkiers.github.io/clusTCR/docs/clustering/how-to-use.html">Usage</a>)</li>
</ul>
<h3>Results</h3>
<p><img alt="New distribution of TCR p gen for data from TCRdb" src="./images/TCRdb_pgen.png">
New distribution of TCR p gen for data from TCRdb</p>
<h1>Horizontal gene transfer</h1>
<h3>Outline</h3>
<ul>
<li>[PalmDB](https://github.com/rcedgar/palmdb}</li>
<li>[x]  download data from the paper</li>
<li>[ ]  can we use their code for our purposes?</li>
</ul>
  </div><!-- /.entry-content -->
</section>
  </div>

  <script src="https://code.jquery.com/jquery-3.6.0.slim.min.js" integrity="sha256-u7e5khyithlIdTpu22PHhENmPcRdFiHRjhAuHcs05RI=" crossorigin="anonymous"></script>
  <script src="https://stackpath.bootstrapcdn.com/bootstrap/4.5.0/js/bootstrap.min.js" integrity="sha384-OgVRvuATP1z7JjHLkuOU7Xw704+h835Lr+6QL9UvYjZE3Ipu6Tp75j7Bh/kR0JKI" crossorigin="anonymous"></script>
  <script src="https://cdn.jsdelivr.net/npm/popper.js@1.16.0/dist/umd/popper.min.js" integrity="sha384-Q6E9RHvbIyZFJoft+2mJbHaEWldlvI9IOYy5n3zV9zzTtmI3UksdQRVvoxMfooAo" crossorigin="anonymous"></script>
</body>

</html>